### ISSUE TITLE
menus missing after install on Rstudio-server

### ISSUE THREAD

**Issue by urielcarrasquilla** [radiant-rstats/radiant#131]
Ubuntu 16.04.6 LTS
R 3.6.1
Access to Radiant via browser using Rstudio-server

Messages received:
> radiant::radiant()

Starting radiant in the default browser ...

Use radiant::radiant_viewer() in Rstudio to open radiant in the Rstudio viewer
or radiant::radiant_window() in Rstudio to open radiant in an Rstudio window

Listening on http://127.0.0.1:7682

However, the Menus on the left pane where the datasets can be uploaded are missing.
I suspect the "git clone" of the Radiant source is not in the correct location.
I put it under /home/rstudio  (same place where R directory is located for all of the R packages in use by user=rstudio).

**Comment by vnijs** [comment]
Please review the "Installing R, Rstudio, and Radiant on Linux" section at the link below. You don't need to clone the sources if you are running from Rstudio. If all ubuntu dependencies are available, running the below should get you the latest version of all relevant packages.

```
install.packages("radiant.update", repos = "https://radiant-rstats.github.io/minicran/")
radiant.update::radiant.update()
```
https://radiant-rstats.github.io/docs/install.html

**Comment by urielcarrasquilla** [comment]
Thank you for your reply.  I ran the two commands.
When I do the radiant::radiant() I still don't see anything under the top
menu: "Radiant Data Design BASICS MODEL Multivariate Report".
A picture:
[image: image.png]
Uri Carrasquilla
Senior Data Scientist / Planner
"adventure is just bad planning" - Roald Amundsen
(561) 900-8296 <555-555-5555>
ucarrasquilla@digitalocean.com

**Comment by vnijs** [comment]
Can you post the screenshot again? I can't see it on github. Also, are there any messages shown in Rstudio or in the Developer Console of the web-browser?

**Comment by urielcarrasquilla** [comment]
Before I do that let me share some findings.
When I reinstalled all the packages as per your instructions, I noticed
that some of them are being reported with a non-zero exit.  I went back to
see why and in some cases it is due to a prerequisite being missing such as
openssl, igraph and rgdal.  When I did an install.packages and re-installed
the ones with non-zero exit, they completed without the warning.  I may
have missed some.  As a reference here is the list of all the packages that
were selected for updates:
install.packages(c("radiant", "radiant.basics", "radiant.data",
"radiant.design", "radiant.model", "shinyAce", "shinyFiles", "AlgDesign",
"BH", "DBI", "DEoptimR", "DiagrammeR", "DiceDesign", "Formula",
"GPArotation", "GPfit", "Gmedian", "Hmisc", "ISOcodes", "KernSmooth",
"Matrix", "MatrixModels", "Metrics", "ModelMetrics", "NLP",
"NeuralNetTools", "PerformanceAnalytics", "Quandl", "R.cache",
"R.methodsS3", "R.oo", "R.utils", "R6", "RColorBrewer", "RPostgreSQL",
"RSQLite", "RSelenium", "RSpectra", "Rcpp", "RcppArmadillo", "RcppEigen",
"RcppParallel", "RgoogleMaps", "RhpcBLASctl", "Rook", "Rttf2pt1",
"SQUAREM", "SnowballC", "SparseM", "StanHeaders", "TH.data", "TTR", "XML",
"abind", "acepack", "alphavantager", "anytime", "arm", "arrow", "askpass",
"assertthat", "backports", "bayesplot", "binman", "bit", "bit64", "bitops",
"blob", "bookdown", "boot", "brew", "caTools", "callr", "car", "carData",
"caret", "cellranger", "checkmate", "class", "classInt", "cli", "clipr",
"clisymbols", "clustMixType", "cluster", "coda", "codetools", "coin",
"collections", "colorspace", "colourpicker", "colourvalues", "commonmark",
"config", "covr", "crayon", "crosstalk", "cyclocomp", "data.table",
"dbplyr", "desc", "devtools", "dials", "dichromat", "digest", "downloader",
"dtplyr", "dygraphs", "e1071", "ellipsis", "evaluate", "extrafont",
"extrafontdb", "fansi", "farver", "fastmap", "fastmatch", "flexdashboard",
"forcats", "foreach", "forecast", "foreign", "forge", "formatR",
"fracdiff", "fs", "furrr", "future", "future.apply", "gargle", "gbm",
"gdalUtils", "gdtools", "generics", "geojsonsf", "gganimate", "ggforce",
"ggraph", "ggrepel", "ggridges", "gh", "gifski", "git2r", "gitgadget",
"glmnet", "globals", "glue", "googleAuthR", "googlePolylines", "gower",
"graphlayouts", "gridExtra", "gtable", "gtools", "gtrendsR", "hardhat",
"haven", "here", "hexbin", "highr", "hms", "htmlTable", "htmltools",
"htmlwidgets", "httpuv", "httr", "hunspell", "igraph", "iml", "infer",
"influenceR", "ini", "inline", "inum", "ipred", "isoband", "iterators",
"janeaustenr", "jpeg", "jsonify", "keras", "labeling", "languageserver",
"later", "lattice", "latticeExtra", "lava", "lazyeval", "lda", "leafem",
"leaflet", "leaflet.providers", "leafpop", "leafsync", "lfe", "lhs",
"libcoin", "lifecycle", "lime", "lintr", "listenv", "lme4", "lmtest",
"loo", "lwgeom", "magick", "magrittr", "mapdeck", "maptools", "mapview",
"matrixStats", "memoise", "mgcv", "mime", "miniUI", "minqa", "mnormt",
"modelr", "modeltools", "multcomp", "munsell", "network", "nlme", "nloptr",
"numDeriv", "odbc", "openssl", "openxlsx", "pROC", "packrat", "padr",
"pageviews", "parsnip", "party", "partykit", "pbkrtest", "pdp", "pillar",
"pkgbuild", "pkgconfig", "pkgload", "plogr", "plotly", "plyr", "polyclip",
"polycor", "pool", "praise", "prediction", "prettyunits", "processx",
"prodlim", "progress", "promises", "proxyC", "ps", "purrr", "pwr",
"quadprog", "quanteda", "quantmod", "quantreg", "r2d3", "randomForest",
"rapidjsonr", "rappdirs", "rcmdcheck", "recipes", "rematch", "rematch2",
"remotes", "renv", "repr", "reprex", "reshape2", "reticulate", "rex",
"rgdal", "rgeos", "rgexf", "riingo", "rio", "rjson", "rlang", "robustbase",
"roxygen2", "rpart", "rpart.plot", "rprojroot", "rsample", "rsconnect",
"rstan", "rstanarm", "rstantools", "rstudioapi", "rtweet", "rversions",
"rvest", "sandwich", "satellite", "selectr", "semver", "servr",
"sessioninfo", "sfheaders", "shape", "shinydashboard", "shinyjs",
"shinystan", "shinythemes", "slam", "sna", "sourcetools", "sp", "spacyr",
"sparklyr", "sparkxgb", "spatialwidget", "stargazer", "stars",
"statnet.common", "stm", "stopwords", "stringdist", "stringr",
"strucchange", "styler", "survival", "svglite", "sys", "systemfonts",
"tensorflow", "testthat", "tfestimators", "tfruns", "threejs", "tidygraph",
"tidymodels", "tidyposterior", "tidypredict", "tidyquant", "tidyselect",
"tidytext", "timeDate", "timetk", "tinytex", "tm", "tmap", "tmaptools",
"tokenizers", "topicmodels", "tseries", "tune", "tweenr", "units", "urca",
"usethis", "utf8", "uuid", "vctrs", "vip", "viridis", "viridisLite",
"visNetwork", "wdman", "webshot", "whisker", "withr", "wordcloud",
"workflows", "xaringan", "xfun", "xml2", "xmlparsedata", "xopen", "xptr",
"xtable", "xts", "yaImpute", "yaml", "yardstick", "zeallot", "zip", "zoo"),
lib = "/home/rstudio/R/x86_64-pc-linux-gnu-library/3.6", repos = "
https://rsm-compute-01.ucsd.edu:4242/rsm-msba/__linux__/bionic/latest",
type = "source", INSTALL_opts = "")

When I start Rstudio-server and run radiant::rediant() I get the following
messages:

Starting radiant in the default browser ...

Use radiant::radiant_viewer() in Rstudio to open radiant in the Rstudio viewer
or radiant::radiant_window() in Rstudio to open radiant in an Rstudio window
Listening on http://127.0.0.1:6580


At this point I go to the window opened by radiant and I find the
following in the display:

Radiant Data Design Basics Model Multivariate  Report

This is followed by the icons for saving power ? (for help)


All of them are grey so I can't select anything.









Uri Carrasquilla
Senior Data Scientist / Planner
"adventure is just bad planning" - Roald Amundsen
(561) 900-8296 <555-555-5555>
ucarrasquilla@digitalocean.com

**Comment by urielcarrasquilla** [comment]
[image: image.png]
Uri Carrasquilla
Senior Data Scientist / Planner
"adventure is just bad planning" - Roald Amundsen
(561) 900-8296 <555-555-5555>
ucarrasquilla@digitalocean.com

**Comment by urielcarrasquilla** [comment]
![image](https://user-images.githubusercontent.com/39807520/79394034-89863880-7f44-11ea-9d7a-802218a6a1c8.png)

**Comment by vnijs** [comment]
I'm 99% sure you are still missing some linux dependencies. On Windows and macOS these dependencies are automatically include. This is not possible on Linux however. Did you follow the dependency instructions for linux listed on the page below?

https://radiant-rstats.github.io/docs/install.html

We run radiant in linux servers and through Rstudio on Ubuntu 18.04 without any issues. I'm not sure what shinyapps.io uses but likely 16 or 18. See demo below.

Free-tier instance on shinyapps.io: https://vnijs.shinyapps.io/radiant/

**Comment by urielcarrasquilla** [comment]
When Running:
https://raw.githubusercontent.com/radiant-rstats/minicran/gh-pages/update.R

#################################################
Updating previously installed packages
#################################################

Error in update.packages(lib.loc = .libPaths()[1], ask = FALSE, repos =
repos_run,  :
  object 'repos_run' not found
>
> # err <- c("yaml", "Matrix", "radiant", "radiant.model")
> # err <- c("yaml", "Matrix")
> # err <- c()
> err <- err[!err %in% installed.packages()]
Error: object 'err' not found
Uri Carrasquilla
Senior Data Scientist / Planner
"adventure is just bad planning" - Roald Amundsen
(561) 900-8296 <555-555-5555>
ucarrasquilla@digitalocean.com

**Comment by vnijs** [comment]
Ah. You found a bug in the update script! Thanks for reporting @urielcarrasquilla. Can you try again now? 

BTW I have Rstudio Package Manager running at the url below. However, this only provides binaries for Ubuntu 18.04

https://rsm-compute-01.ucsd.edu:4242/client/

**Comment by vnijs** [comment]
Also, you may be interested in the docker container I have that includes shiny-server, rstudio-server and radiant

Images: https://hub.docker.com/r/vnijs/radiant
Docker and launch files: https://github.com/radiant-rstats/docker

**Comment by urielcarrasquilla** [comment]
I managed to run the refreshed script.
Followed the remaining steps in the install.html link you provided.
I did an "rstudio-server stop" followed by a start.
From the RStudio Server issued:

Stopped Radiant


Attaching package: ‘radiant.data’
The following objects are masked from ‘package:lubridate’:

    month, wday
The following object is masked from ‘package:ggplot2’:

    diamonds
> radiant::radiant()Starting radiant in the default browser ...

Use radiant::radiant_viewer() in Rstudio to open radiant in the Rstudio viewer
or radiant::radiant_window() in Rstudio to open radiant in an Rstudio window
Listening on http://127.0.0.1:6827

Stopped Radiant


>

No Luck. I am still not getting the left hand side Pane.



Uri Carrasquilla
Senior Data Scientist / Planner
"adventure is just bad planning" - Roald Amundsen
(561) 900-8296 <555-555-5555>
ucarrasquilla@digitalocean.com

**Comment by vnijs** [comment]
Wow. I can't see any issues in the install process. However, (1) I can't reproduce the issue on any computer I have access to (see e.g., screenshot below for Rstudio server) and (2) there are no error messages in Rstudio or in the developer log for me to work from.

Is the docker container an option for you? Or is there any way for me to temporarily access the server to see if I can spot something? My work email is vnijs@ucsd.edu

![image](https://user-images.githubusercontent.com/2998956/79409420-c48f6880-7f52-11ea-93c6-3ebc7326ab27.png)

**Comment by vnijs** [comment]
I did also just update radiant on CRAN. Could you just try:

```
install.packages("radiant.data")
radiant.data::radiant.data()
```

I just tried this on rstudio cloud and it is running fine

![image](https://user-images.githubusercontent.com/2998956/79411293-62853200-7f57-11ea-84d0-6508061e1348.png)

**Comment by vnijs** [comment]
radiant also installed and ran without a hitch in rstudio cloud. So nice to have binary packages from Rstudio so it doesn't take HOURS to install packages :)

```
install.packages("radiant")
radiant::radiant()
```

**Comment by urielcarrasquilla** [comment]
Hi Vincent.

Installed docker and it is working now.
This was done on Ubuntu version 18 04 4 LTS

Got the following error messages after running from bash cli:
docker run --rm -p 8080:8080 -p 8787:8787 -v ~:/home/jovyan vnijs/radiant

Status: Downloaded newer image for vnijs/radiant:latest
2020-04-16 10:16:05,862 INFO supervisord started with pid 1
2020-04-16 10:16:06,864 INFO spawned: 'shinyserver' with pid 8
2020-04-16 10:16:06,866 INFO spawned: 'rserver' with pid 9
2020-04-16 10:16:06,868 INFO spawned: 'userconf' with pid 10
2020-04-16 10:16:06,877 INFO success: rserver entered RUNNING state,
process has stayed up for > than 0 seconds (startsecs)
2020-04-16 10:16:06,877 INFO success: userconf entered RUNNING state,
process has stayed up for > than 0 seconds (startsecs)
2020-04-16 10:16:06,890 INFO exited: rserver (exit status 0; expected)
2020-04-16 10:16:06,891 INFO exited: userconf (exit status 0; expected)
2020-04-16 10:16:08,228 INFO success: shinyserver entered RUNNING state,
process has stayed up for > than 1 seconds (startsecs)

Then, went to browser and pointed to the server where docker installed.
Added :8080 to the end of the URL.  Nothing found.
Added: 8787 and got sign on panel for RStudio.  How do I find out the
user_id and password?

Thank you.


Uri Carrasquilla
Senior Data Scientist / Planner
"adventure is just bad planning" - Roald Amundsen
(561) 900-8296 <555-555-5555>
ucarrasquilla@digitalocean.com

**Comment by vnijs** [comment]
For radiant, use 8181:8181 instead. Username: jovyan, Password: rstudio

`docker run --rm -p 8181:8181 -p 8787:8787 -v ~:/home/jovyan
 vnijs/radiant`

**Comment by urielcarrasquilla** [comment]
That worked for me.  8181 is the port.
I can now load files from my laptop.  Thank you.
I have a file system that is hosted via Apache.  Is there any way I can
point radiant to it instead of only being able to look at my laptop file
system?
Thank you.

Uri Carrasquilla
Senior Data Scientist / Planner
"adventure is just bad planning" - Roald Amundsen
(561) 900-8296 <555-555-5555>
ucarrasquilla@digitalocean.com

**Comment by vnijs** [comment]
If you start Radiant from RStudio on a server you should have access to the filesystem on the server as well. You can also load any data you have access to through R > Rmarkdown if you can (1) store it in a data.frame (e.g., my_data) and then (2) use `register("my_data")` so the data shows up from the dataset dropdowns.

If you prefer using Radiant through shiny-server (i.e., port 8181) then adding the options below to your .Rprofile file should allow you to look at files in your home-directory on the server. This last part is not that well tested because I have most students use Radiant through Rstudio.

If you want to have a quick zoom call about what you are trying to achieve, reach out to me through vnijs@ucsd.edu

```
options(radiant.maxRequestSize = -1)
options(radiant.report = TRUE)
options(radiant.shinyFiles = TRUE
```
